### PR TITLE
SnippetTestCase moved to eo-runtime

### PR DESCRIPTION
**Pull request by yegor256**
See #2721

I also introduced `WeAreOnline` from "jping": https://github.com/yegor256/jping

---

## PR-Codex overview
This PR focuses on code refactoring and test improvements. 

### Detailed summary
- Code refactoring: Removed unnecessary exception declarations in test setup methods.
- Test improvements: Replaced custom online condition with a third-party library for better test organization.
- Added a new test dependency for testing purposes.
- Updated dependencies in the pom.xml file.
- Minor code changes and imports cleanup.

> The following files were skipped due to too many changes: `eo-maven-plugin/src/main/java/org/eolang/maven/DiscoverMojo.java`, `eo-maven-plugin/src/test/java/org/eolang/maven/objectionary/OyIndexedTest.java`, `eo-maven-plugin/src/test/java/org/eolang/maven/objectionary/ObjectsIndexTest.java`, `eo-maven-plugin/src/test/java/org/eolang/maven/dependencies/DcsWithRuntimeTest.java`, `eo-maven-plugin/src/test/java/org/eolang/maven/AssembleMojoTest.java`, `eo-maven-plugin/src/test/java/org/eolang/maven/objectionary/OyRemoteTest.java`, `eo-maven-plugin/pom.xml`, `eo-runtime/pom.xml`, `eo-maven-plugin/src/test/java/org/eolang/maven/ProbeMojoTest.java`, `eo-maven-plugin/src/main/java/org/eolang/maven/SafeMojo.java`, `eo-maven-plugin/src/test/java/org/eolang/maven/it/SnippetTestCase.java`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by yegor256** [comment]
@maxonfjvipon please, take a look

**Comment by maxonfjvipon** [comment]
@yegor256 what's the point of replacing `OnlineCondition` with `WeAreOnline`?

**Comment by yegor256** [comment]
@maxonfjvipon we need in both eo-maven-plugin and in eo-runtime. Thus, it should be implemented somewhere outside of them both. I decided to make it a standalone library. Maybe some other projects will need it too.

File: eo-runtime/pom.xml [hunk 1]
```diff
@@ -104,6 +110,34 @@ SOFTWARE.
       <!-- version from parent POM -->
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>com.yegor256</groupId>
+      <artifactId>jping</artifactId>
+      <!-- version from parent POM -->
```
Comment (maxonfjvipon) [inline eo-runtime/pom.xml:116]: @yegor256 all dependencies in `eo-runtime` pom.xml must have "test" scope. Build fails otherwise

Comment (yegor256) [reply eo-runtime/pom.xml:116]: @maxonfjvipon fixed in [aa6a32a](https://github.com/objectionary/eo/pull/2725/commits/aa6a32af380c450a38599a85928e93456b928649)

File: eo-runtime/pom.xml [hunk 2]
```diff
@@ -104,6 +110,35 @@ SOFTWARE.
...
+    <dependency>
+      <groupId>com.yegor256</groupId>
+      <artifactId>farea</artifactId>
+      <version>0.0.13</version>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.eolang</groupId>
+      <artifactId>jucs</artifactId>
+      <!-- version from parent POM -->
```
Comment (maxonfjvipon) [inline eo-runtime/pom.xml:140]: @yegor256 here too

Comment (yegor256) [reply eo-runtime/pom.xml:140]: @maxonfjvipon thanks, fixed [0aa2e0a](https://github.com/objectionary/eo/pull/2725/commits/0aa2e0a908fc453cce393d3eeb5190e00af8b861)

**Comment by yegor256** [comment]
@maxonfjvipon it's ready

**Review by maxonfjvipon — APPROVED** [review]
@yegor256 looks good to me

**Comment by yegor256** [comment]
@rultor merge

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**